src/rgw/rgw_process.cc

```diff
@@ -87,12 +90,83 @@ void RGWProcess::RGWWQ::_process(RGWRequest *req, ThreadPool::TPHandle &) {
   process->req_throttle.put(1);
   perfcounter->inc(l_rgw_qactive, -1);
 }
+bool rate_limit(rgw::sal::Store* store, req_state* s) {
+  // we dont want to limit health check or system or admin requests
+  const auto& is_admin_or_system = s->user->get_info();
+  if (s->op_type ==  RGW_OP_GET_HEALTH_CHECK || is_admin_or_system.admin || is_admin_or_system.system)
+    return false;
+  std::string userfind;
+  RGWQoSInfo global_user;
+  RGWQoSInfo global_bucket;
+  RGWQoSInfo global_anon;
+  RGWQoSInfo* bucket_qos;
+  RGWQoSInfo* user_qos;
+  store->get_qos(global_bucket, global_user, global_anon);
```

Comment:
what if all qos info object has a "disabled" flag, why not bail out here?

Reply:
because there are the global values and the per bucket and user values which should be take in place anyway (the global will take in place only if the specific values are disabled).